packages/snap/src/utils/getGuardianInfo.ts

```diff
@@ -0,0 +1,50 @@
+import type {
+  ProviderMeta,
+  ZkCertRegistered,
+} from '@galactica-net/galactica-types';
+import type { Address } from 'viem';
+import { getContract } from 'viem';
+
+import { getWalletClient } from './getWalletClient';
+import { guardianRegistryABI } from '../config/abi/guardianRegistry';
+import { kycRecordRegistryABI } from '../config/abi/kycRecordRegistry';
+
+export const getGuardianInfo = async (cert: ZkCertRegistered) => {
+  try {
+    const wc = await getWalletClient();
```

Comment:
I am not sure if this always works because the wallet client created here might be on another chain than the zkCert.
In my opinion it makes sense to ready the chainID from ZkCertRegistered and switch to it before getting the walletClient.
There is a function for it in utils.

Reply:
This line is not present anymore in the latest version. But I think we need something like `await switchChain(zkCert.registration.chainID, ethereum);` because the guardian registry is only on Galactica and not on another EVM Metamask might be connected to at the moment of execution.